core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -674,7 +675,10 @@ bool core_mmu_divide_block(struct core_mmu_table_info *tbl_info,
 		return false;
 
 	entry = (uint64_t *)tbl_info->table + idx;
-	assert((*entry & DESC_ENTRY_TYPE_MASK) == BLOCK_DESC);
+	assert(!*entry || (*entry & DESC_ENTRY_TYPE_MASK) == BLOCK_DESC);
+
+	/* We need to flush TLBs only if there already was some mapping */
+	flush_tlb = *entry;
```

Comment:
No. C99 standard states the following:

> 6.3.1.2 Boolean type
> When any scalar value is converted to _Bool, the result is 0 if the value compares equal
> to 0; otherwise, the result is 1.